internal/resources/metal/vlan/models.go

```diff
@@ -21,29 +21,34 @@ type DataSourceModel struct {
 	AssignedDevicesIds types.List   `tfsdk:"assigned_devices_ids"`
 }
 
-func (m *DataSourceModel) parse(vlan *packngo.VirtualNetwork) (d diag.Diagnostics) {
-	m.ID = types.StringValue(vlan.ID)
-	m.VlanID = types.StringValue(vlan.ID)
-	m.Vxlan = types.Int64Value(int64(vlan.VXLAN))
+func (m *DataSourceModel) parse(vlan *metalv1.VirtualNetwork) (d diag.Diagnostics) {
+	m.ID = types.StringValue(vlan.GetId())
+	m.VlanID = types.StringValue(vlan.GetId())
+	m.Vxlan = types.Int64Value(int64(vlan.GetVxlan()))
 	m.Facility = types.StringValue("")
 
-	if vlan.Description != "" {
-		m.Description = types.StringValue(vlan.Description)
+	if vlan.GetDescription() != "" {
+		m.Description = types.StringValue(vlan.GetDescription())
 	}
 
-	if vlan.Project.ID != "" {
-		m.ProjectID = types.StringValue(vlan.Project.ID)
+	if vlan.AdditionalProperties["project"] != nil {
+		project := vlan.AdditionalProperties["project"].(map[string]interface{})
+		project_id := project["id"].(string)
+		m.ProjectID = types.StringValue(project_id)
 	}
 
 	if vlan.Facility != nil {
-		m.Facility = types.StringValue(strings.ToLower(vlan.Facility.Code))
-		m.Metro = types.StringValue(strings.ToLower(vlan.Facility.Metro.Code))
+		facility_code := vlan.Facility.AdditionalProperties["code"].(string)
+		metro := vlan.Facility.AdditionalProperties["metro"].(map[string]interface{})
```

Comment:
This _could_ be improved by updating the OpenAPI spec but probably not worth it in this case.